### PR TITLE
feat: unit tests for Katas

### PR DESCRIPTION
**Pull request by martin728**

### CODE REVIEW

File: reflection/Katas/KatasTest/Tests.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace KatasTest
+{
+    [TestFixture]
+    public class Tests
+    {
+        private Katas.Katas _fizzbuzz;
```
Comment (drudevych) [inline reflection/Katas/KatasTest/Tests.cs]: rename it to something like "_katas" since you work with different algorithms using this field, not only FizzBuzz()

File: reflection/Katas/KatasTest/Tests.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        }
+        
+        //FizzBuzz
+        [TestCase(1, "1")]
+        [TestCase(3, "Fizz")]
+        [TestCase(5, "Buzz")]
+        [TestCase(15, "FizzBuzz")]
+        [TestCase(13, "13")]
+        [TestCase(30, "FizzBuzz")]
+        [TestCase(98, "98")]
```
Comment (drudevych) [inline reflection/Katas/KatasTest/Tests.cs:24]: would be good to add tests cases when input number is out of 1-100 range. You should also modify FizzBuzz() method to handle such cases

Comment (drudevych) [reply reflection/Katas/KatasTest/Tests.cs:24]: you can create another test method and name it something like "FizzBuzz_WithOutOfRangeInputs_ThrowsException"

File: reflection/Katas/KatasTest/Tests.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        [TestCase(12, "Even")]
+        [TestCase(13, "13")]
+        [TestCase(14, "Even")]
+        [TestCase(15, "Odd")]
+        [TestCase(16, "Even")]
+        [TestCase(17, "17")]
+        [TestCase(18, "Even")]
+        [TestCase(19, "19")]
+        [TestCase(20, "Even")]
+        public void CheckNumber_ReturnsExpectedResult(int num, string expectedResult)
```
Comment (drudevych) [inline reflection/Katas/KatasTest/Tests.cs:64]: the same here, as an improvement you can add tests for out of range cases (1-100)

File: reflection/Katas/Katas/Katas.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                    isPrime = false;
+                    break;
+                }
+            }
+
+            if (isPrime)
+            {
+                return num.ToString();
+            }
+            else if (num % 2 == 0)
```
Comment (drudevych) [inline reflection/Katas/Katas/Katas.cs]: simplify last 2 "else if" blocks with ternary operator, e.g.:
`return num % 2 == 0 ? "Even" : "Odd";`

File: reflection/Katas/Katas/Katas.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        
+        //The Leap Year Kata
+        public bool IsLeapYear(int year)
+        {
+            if (year % 4 != 0)
+            {
+                return false;
+            }
+            else if (year % 100 != 0)
```
Comment (drudevych) [inline reflection/Katas/Katas/Katas.cs]: remove redundant else

File: reflection/Katas/Katas/Katas.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            {
+                return true;
+            }
+            else if (year % 400 == 0)
```
Comment (drudevych) [inline reflection/Katas/Katas/Katas.cs]: the same here, you can simplify these blocks with single ternary operator

File: reflection/Katas/Katas/Katas.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            {
+                return true;
+            }
+            else
+            {
+                return false;
```
Comment (drudevych) [inline reflection/Katas/Katas/Katas.cs]: in general, it's a good practice to have a single "exit"(return) in your methods, this simplifies the code. Try to change this and previous methods to have only one return statement, you can provide a local variable, that will store the value you want to return and set it depending on condition.

File: reflection/Katas/KatasTest/Tests.cs [hunk 8]
```diff
@@ -25,12 +25,22 @@ public void Setup()
...
         {
             // Act
-            string result = _fizzbuzz.FizzBuzz(number);
+            string result = _katas.FizzBuzz(number);
             
             // Assert
             Assert.AreEqual(expected, result);
         }
         
+        [Test]
```
Comment (drudevych) [inline reflection/Katas/KatasTest/Tests.cs:34]: add at least 2 TestCases with <= 0 and >100 using [TestCase] attribute as you did in other tests

File: reflection/Katas/Katas/Katas.cs [hunk 9]
```diff
@@ -10,11 +10,17 @@ public class Katas
             number % 15 == 0 ? "FizzBuzz" :
             number % 3 == 0 ? "Fizz" :
             number % 5 == 0 ? "Buzz" :
+            number > 100 || number < 0 ? throw new ArgumentOutOfRangeException() :
             number.ToString();
 
         //The OddEven Kata
         public string CheckNumber(int num)
         {
+            if (num < 0 || num > 100)
```
Comment (drudevych) [inline reflection/Katas/Katas/Katas.cs]: it should be <= 0, I believe

File: reflection/Katas/Katas/Katas.cs [hunk 10]
```diff
@@ -10,11 +10,17 @@ public class Katas
             number % 15 == 0 ? "FizzBuzz" :
             number % 3 == 0 ? "Fizz" :
             number % 5 == 0 ? "Buzz" :
+            number > 100 || number < 0 ? throw new ArgumentOutOfRangeException() :
```
Comment (drudevych) [inline reflection/Katas/Katas/Katas.cs]: <= 0

File: reflection/Katas/KatasTest/Tests.cs [hunk 11]
```diff
@@ -55,12 +65,22 @@ public void FizzBuzz_WithVariousInputs_ReturnsExpectedOutput(int number, string
...
         {
             //Act
-            string result = _fizzbuzz.CheckNumber(num);
+            string result = _katas.CheckNumber(num);
             
             //Assert
             Assert.AreEqual(expectedResult, result);
         }
         
+        [Test]
```
Comment (drudevych) [inline reflection/Katas/KatasTest/Tests.cs:73]: the same here, add 2 test cases to check both edges

**Review by drudevych — APPROVED** [review]
